src/main/kotlin/org/rust/lang/core/resolve2/ModCollectorBase.kt

```diff
@@ -151,12 +151,30 @@ class ModCollectorBase private constructor(
             name = def.name ?: return,
             body = def.macroBody ?: return,
             bodyHash = def.bodyHash,
-            hasMacroExport = def.hasMacroExport,
-            hasLocalInnerMacros = def.hasMacroExportLocalInnerMacros
+            hasMacroExport = HAS_MACRO_EXPORT_PROP.getByStub(def, crate),
+            hasLocalInnerMacros = HAS_MACRO_EXPORT_LOCAL_INNER_MACROS_PROP.getByStub(def, crate)
         )
         visitor.collectMacroDef(defLight)
     }
 
+    private val RsAttributeOwnerStubBase<out RsDocAndAttributeOwner>.hasMacroUse: Boolean
+        get() = mayHaveMacroUse && psi.getQueryAttributes(crate, this).hasAttribute("macro_use")
```

Comment:
Potentially we can now introduce `hasMacroUse` for `RsModItemStub`, `RsModDeclItemStub` and `RsExternCrateItemStub` using their `HAS_MACRO_USE_PROP`, but I am not sure if it worth it